### DISCUSSION TITLE
pull_request webhooks contain inconsistent data

### DISCUSSION THREAD

**Discussion by jeyq** [community/community#28877] [Apps, API and Webhooks]
Hi,

we're seeing inconsistent data in pull request webhooks. Sometimes when we receive a `pull_request` webhook with `closed` action, the `closed_at` field in the payload is `null`. Also, it seems like the `updated_at` timestamp in the payload for pull_request doesn't account for the change that triggered this webhook. And, the `state` field also is `open` even if the `action=closed` In these cases it also seems that when we then fetch the PR information from the GraphQL API immediately, it also doesn't represent the latest state info. After a while, we then get the up to date information from the API.

Most of the times the information is consistent but it seems that there's still non-trivial amount of webhooks we get that have this consistency problem.

Is there something we could do to avoid this and ensure we get the latest up to date information about the pull request in the webhooks?

**Comment by pbstriker38** [comment]
This is also affecting us. Any updates on this?

**Comment by fintap-nsossonko** [comment]
This is affecting us now and hindering our ability to seamlessly run our external pipeline. Any updates?
